### ISSUE TITLE
Crash in Door.PushCharactersAway

### ISSUE THREAD

**Issue by Regalis11** [Regalis11/Barotrauma#497]
The position of a character or the door seems to be NaN for some reason.

[crashreport.txt](https://github.com/Regalis11/Barotrauma/files/2205382/crashreport.txt)

**Comment by NilanthAnimosus** [comment]
I've seen this before and had no idea why, I tried to fix it myself in my mod by changing  the code but simply broke the function entirely (It sucked characters into the door, pressed them down then crushed them effectively) and ended up reverting. though that was some weeks ago (Before 0.8.1.3) its in the exact same place.

Not too sure what causes it either.

**Comment by Regalis11** [comment]
This could be related to f13b1b4 and/or 9e69216, because ScreenToWorld and WorldToScreen always returned {NaN, NaN} before the camera transforms were updated and clients with command permissions were able to spawn items and characters at {NaN, NaN}.

**Comment by NilanthAnimosus** [comment]
Would the above effect dedicated servers? As far as I know for the crash report I have it is from a dedictated server.

**Comment by Regalis11** [comment]
It could affect dedicated servers too, some of the console commands use the hosts cursor position (which is always at the center of the main sub) and positions sent by clients.

**Comment by NilanthAnimosus** [comment]
I thought character simpositions were server-side, and the item positions also server-side. as I understood the code the clients send their inputs and the server processes them, and returns their character state/position and other data.

**Comment by Regalis11** [comment]
You're right, but console commands like spawn and spawnitem may use the cursor position sent by a client, which wasn't validated in any way before 9e69216. So a clients could spawn a character or an item at {NaN, NaN} and crash the server.

**Comment by NilanthAnimosus** [comment]
Alright, that makes more sense now. I hadn't thought of it that way. that could be related, Guess we will know in time based on the analytics / reports.
